ott/core/gw_barycenter.py

```diff
@@ -0,0 +1,323 @@
+from functools import partial
+from typing import Any, Dict, NamedTuple, Optional, Sequence, Tuple, Union
+
+import jax
+import jax.numpy as jnp
+
+from ott.core import (
+    fixed_point_loop,
+    gromov_wasserstein,
+    problems,
+    quad_problems,
+    was_solver,
+)
+from ott.core.bar_problems import GWBarycenterProblem
+from ott.geometry import geometry, pointcloud
+
+__all__ = ["GWBarycenterState", "GromovWassersteinBarycenter"]
+
+
+class GWBarycenterState(NamedTuple):
+  """Holds the state of the :class:`ott.core.bar_problems.GWBarycenterProblem`.
+
+  Args:
+    x: Barycenter features of shape ``[B, D_f]``. Only used in the fused case.
+    c: Barycenter cost matrix of shape ``[B, B]``.
+    a: Weights of the barycenter of shape ``[B,]``.
+    errors: Array of shape
+      ``[max_iter, num_measures, quad_max_iter, lin_outer_iter]`` containing
+      the GW errors at each iteration.
+    costs: Array of shape ``[max_iter,]`` containing the cost at each iteration.
+    gw_convergence: Array of shape ``[max_iter,]`` containing the convergence
+      of all GW problems at each iteration.
+  """
+  x: Optional[jnp.ndarray] = None
+  c: Optional[jnp.ndarray] = None
```

Comment:
i feels c should come first, and maybe call it cost to be clear that this is not a vector or a point cloud

Reply:
Renamed to `cost`.